### ISSUE TITLE
Scratch Lab

### ISSUE THREAD

**Issue by areeq0325** [ScratchAddons/ScratchAddons#1584]
**Describe the bug**
Extensions not loading on lab.scratch.mit.edu pages

**To Reproduce**
Go to any Scratch Lab page

**Expected behavior**
Addons work

**Screenshots**
![image](https://user-images.githubusercontent.com/69761212/107969693-18f1e300-6f7e-11eb-9343-06fc2ef0cf95.png)
I have dark mode on

**Device (please complete the following information):**
Windows 10 20h2, Edge 88.0.705.68

EDIT: lab.scratch.MIT.edu, not lab.scratch.LIT.edu

**Comment by RedGuy12** [comment]
I don't think it was intended to work on the scratch lab. this would be a pretty big update since it's a separate domain...
________________________________

**Comment by TheColaber** [comment]
This is intensional and if we did want to add this domain, it would require the user to have to enable it somehow. I think if we add something to allow multiple domains, we could do this.

**Comment by RedGuy12** [comment]
> I think if we add something to allow multiple domains, we could do this.


scratch lab and what else? Turbowarp already has addons built in...

________________________________

**Comment by TheColaber** [comment]
Any new mods that get created as well as llk's and mods.

**Comment by RedGuy12** [comment]
like https://llk.github.io/scratch-gui/ ?
________________________________

**Comment by mxmou** [comment]
Scratch Addons doesn't currently run on any domain other than scratch.mit.edu. This could be changed using optional permissions, but it isn't a bug.

**Comment by WorldLanguages** [comment]
This isn't in any way expected behavior.

**Comment by LankyBox01** [comment]
Uhm, this is kindaaa unrelated to the topic, but, i cannot access scratchlab. Do you need like a special permission or something?

**Comment by RedGuy12** [comment]
> Uhm, this is kindaaa unrelated to the topic, but, i cannot access scratchlab. Do you need like a special permission or something?
@LankyBox01 
there's no homepage for lab.scratch.mit.edu

you have to go to lab.scratch.mit.edu/face or lab.scratch.mit.edu/text

**Comment by LankyBox01** [comment]
> > Uhm, this is kindaaa unrelated to the topic, but, i cannot access scratchlab. Do you need like a special permission or something?
> 
> @LankyBox01
> there's no homepage for lab.scratch.mit.edu
> 
> you have to go to lab.scratch.mit.edu/face or lab.scratch.mit.edu/text

Oh ok, also i made this pull request to add the feature in this issue:
https://github.com/ScratchAddons/ScratchAddons/pull/1587

**Comment by GrahamSH-LLK** [comment]
This is absolutely rejected. Adding a new domain to the manifest disables the extension for all users until they agree to it. We would need serious discussion to do this.

**Comment by RedGuy12** [comment]
@GrahamSH-LLK

1. It could be optional
2. We were discussing it here

**Comment by mxmou** [comment]
Reopening, I don't think this being an optional feature is rejected.

**Comment by apple502j** [comment]
"any way" includes optional permissions.

**Comment by mxmou** [comment]
> "any way" includes optional permissions.

But "this isn't expected behavior" doesn't mean "this will never be added", or does it?

**Comment by WorldLanguages** [comment]
It's still a valid suggestion, I just meant to say this is not a bug. I'm reopening but this does not mean this is planned, or something we'd accept a PR for

**Comment by RedGuy12** [comment]
Dupe of #23
________________________________

**Comment by TheColaber** [comment]
> Dupe of #23

It's not. Read the issue.

**Comment by Barthdry** [comment]
Actually you can try to modify the site that it locates to and rename and change the extension then it will be possible.

**Comment by apple502j** [comment]
The backbone is finished in #6421. Since I doubt Scratch Lab receives updates I will close this as wontfix; it's now way easier to modify yourself.
